runtime/bundles/org.eclipse.core.contenttype/src/org/eclipse/core/internal/content/LazyInputStream.java

```diff
@@ -160,7 +160,7 @@ public long skip(long toSkip) throws IOException {
 		if (toSkip <= 0)
 			return 0;
 		ensureAvailable(toSkip);
-		long skipped = Math.min(toSkip, bufferSize - offset);
+		int skipped = (int) Math.min(toSkip, bufferSize - offset);
```

Comment:
This seems the wrong approach here, we should use a long where needed!

Reply:
I agree that casting to int is not the proper approach here. Either it is not correct that `bufferSize` and/or `offset` are assumed to be in int range, then their types should be changed. Or it is correct that these are both positive int values, then `Math.min` will always return a value within int range (`toSkip` is positive and `bufferSize - offset` is within int range). Then calling `Math.toIntExact` will be safe for the `Math.min` result and could be used instead of an unsafe cast.

Reply:
Since the result of `buffersize - offset` must fit in an `int` (they're both `int`s after all), this cast makes sense. The minimum between a positive long and an int fits in an int.

Reply:
Sorry, I just saw your comment @HeikoKlare . 

We agree that the current code can't throw an exception but since you mentioned `Math::toIntExact`, I changed the code and made it future-ready (one can never be too sure).

Thanks for the tip.

Reply:
@laeubi ok?

Reply:
Just to be sure: have you checked that `offset` being an int is actually correct? Since the code is from 2005, it's worth to check. My proposal was only under the assumption that it is actually correct that we deal with int-range values here. But maybe changing to long is even the proper/better solution here.

Reply:
> Just to be sure: have you checked that offset being an int is actually correct?

This is imposed by the official Java API, you can see it for example in the 2nd parameter of the method `read(byte[] b, int off, int len)` or in the fact the the `mark` field and the method `mark(int readLimit)` also use `int`. Changing `offset` to a `long` only brings more casts with it so it's not an option.

Reply:
I do not understand how those method parameters should be related to the length of an input stream. The first defines the offset within a given _target_ array to be written to, the other defines when a mark position is to be invalidated. None of that prevents a stream with length `> Integer.MAX_VALUE + 1` to be passed to a `LazyInputStream` and a call of `skip(Integer.MAX_VALUE + 1)`, which will never terminate. Simply try:
```java
new LazyInputStream(new InputStream() {
	@Override
	public int read() throws IOException {
		return 0;
	}
}, 100_000).skip((long) Integer.MAX_VALUE + 1);
```
So I think at least `bufferSize` actually needs to be a `long`, no matter how unlikely it is that such a long stream is really used. Or am I missing something?